python/cuml/_thirdparty/sklearn/utils/sparsefuncs.py

```diff
@@ -145,180 +109,64 @@ def inplace_column_scale(X, scale):
-        X.indptr[m + 2:n] += nz_n - nz_m
-        X.indptr[m + 1] = m_start + nz_n
-        X.indptr[n] = n_stop - nz_m
-
-    X.indices = np.concatenate([X.indices[:m_start],
-                                X.indices[n_start:n_stop],
-                                X.indices[m_stop:n_start],
-                                X.indices[m_start:m_stop],
-                                X.indices[n_stop:]])
-    X.data = np.concatenate([X.data[:m_start],
-                             X.data[n_start:n_stop],
-                             X.data[m_stop:n_start],
-                             X.data[m_start:m_stop],
-                             X.data[n_stop:]])
-
-
-def inplace_swap_row(X, m, n):
-    """
-    Swaps two rows of a CSC/CSR matrix in-place.
-
-    Parameters
-    ----------
-    X : CSR or CSC sparse matrix, shape=(n_samples, n_features)
-        Matrix whose two rows are to be swapped.
-
-    m : int
-        Index of the row of X to be swapped.
+    _raise_error_wrong_axis(axis)
 
-    n : int
-        Index of the row of X to be swapped.
-    """
-    if iscsc(X):
-        inplace_swap_row_csc(X, m, n)
-    elif iscsr(X):
-        inplace_swap_row_csr(X, m, n)
+    if iscsr(X):
+        if axis == 0:
+            return _csr_mean_var_axis0(X)
+        else:
+            return _csc_mean_var_axis0(X.T)
+    elif iscsc(X):
+        if axis == 0:
+            return _csc_mean_var_axis0(X)
+        else:
+            return _csr_mean_var_axis0(X.T)
     else:
         _raise_typeerror(X)
 
 
-def inplace_swap_column(X, m, n):
-    """
-    Swaps two columns of a CSC/CSR matrix in-place.
-
-    Parameters
-    ----------
-    X : CSR or CSC sparse matrix, shape=(n_samples, n_features)
-        Matrix whose two columns are to be swapped.
-
-    m : int
-        Index of the column of X to be swapped.
-
-    n : int
-        Index of the column of X to be swapped.
-    """
-    if m < 0:
-        m += X.shape[1]
-    if n < 0:
-        n += X.shape[1]
-    if iscsc(X):
-        inplace_swap_row_csr(X, m, n)
-    elif iscsr(X):
-        inplace_swap_row_csc(X, m, n)
-    else:
-        _raise_typeerror(X)
+ufunc_dic = {
+    'min': np.min,
+    'max': np.max,
+    'nanmin': np.min,
+    'nanmax': np.nanmax
+}
 
 
 def _minor_reduce(X, min_or_max):
-    if min_or_max == 'min':
-        min_or_max = np.min
-    else:
-        min_or_max = np.max
-
+    fminmax = ufunc_dic[min_or_max]
     major_index = np.flatnonzero(np.diff(X.indptr))
-
     # reduceat tries casts X.indptr to intp, which errors
     # if it is int64 on a 32 bit system.
     # Reinitializing prevents this where possible, see #13737
     X = type(X)((X.data, X.indices, X.indptr), shape=X.shape)
 
     value = cpu_np.zeros(len(X.indptr)-1, dtype=X.dtype)
-
     start = X.indptr[0]
     for i, end in enumerate(X.indptr[1:]):
-        value[i] = min_or_max(X.data[start:end])
+        if start != end:
+            value[i] = fminmax(X.data[start:end])
```

Comment:
could you explain a bit about why this had to change? Was it an error found in testing?

Reply:
It wasn't found during testing. I just thought it would be safer to ensure that we don't run the operation on an empty array (case in which a row or column is full of zeroes).